js/utils/helpers.js

```diff
@@ -1850,11 +1850,22 @@
                         if (attrs[0].children[0].type == "link_open") {
                             var iframeHTML = "<iframe ", openingLink = attrs[0].children[0];
                             var enlargeLink, posTop = true, captionClass = "", captionStyle = "", iframeClass = "", iframeStyle = "";
+                            var isYTlink = false, videoText = "";
 
                             // Add all attributes to the iframe
                             openingLink.attrs.forEach(function(attr) {
                                 if (attr[0] == "href") {
+                                    isYTlink  = (attr[1].indexOf("www.youtube.com") != -1 ? true : false);
+                                    
+                                    // Add a class to hide the iframe only if the markdown is a YouTube video
+                                    if(isYTlink)
+                                    {
+                                      iframeHTML += 'class="hide-in-print"' + " ";
```

Comment:
As I mentioned this will cause double `class` attributes on the element. So you need to modify it. 

If you define an iframe with classes this way:
```
::: iframe [CAPTION](https://example.com){link=https://example.com .test .another-class } \n:::
```

One of the attributes that you get in this forEach loop is going to have `attr[0] === "class"`. So you should merge the class that you're adding here and that class together.  

You can see how we're doing this in the video tag. In there we're always attaching a class, so we had to do something similar (Take a look at the `videoClass` attribute in there.)

Reply:
You should change your code to cover more patterns of youtube links. It's also cleaner if in here you just set the `isTYlink` and capture the link value. And move the part that you're defining your "videoText" to be in the same `if (isYTLink)` block that you have below.

So

```javascript
var isYTlink = false, videoURL = "";
openingLink.attrs.forEach(function(attr) {
  if (attr[0] == "href") {
    isYTlink  = (attr[1].match("^(http(s)?:\/\/)?((w){3}.)?youtu(be|.be)?(\.com)?\/.+") != null);
    videURL = attr[1];
  }
  ...
}

...

if (isYTlink) {
  html =  '<span class="video-info-in-print" style="visibility:hidden">';
  htm += 'Note: YouTube video ( ' + attr[1] + ' ) is hidden in print';
  htm += '</span>' + html;
}



```

Reply:
- Made all the changes that were requested. All test cases are passing except for this one: 

`Testing for tuple 0 with row values {id=4000, some_markdown= **date is :**, name=Hank, url= https://www.google.com, some_gene_sequence= GATCGATCGCGTATT, video_col= http://techslides.com/demos/sample-videos/small.mp4, catalog_snapshot_uri=schema:table, catalog_id_uri=schema:table}`

 ```
✗ should check value and isHTML for 'video_col' column
 - Expected '<span class="video-info-in-print" style="visibility:hidden">Note: Video (http://techslides.com/demos/sample-videos/small.mp4) is hidden in print </span><video controls class="-chaise-post-load hide-in-print " height=500 width=600 loop ><source src="http://techslides.com/demos/sample-videos/small.mp4" type="video/mp4"></video>' to be '<video controls class="-chaise-post-load" height=500 width=600 loop ><source src="http://techslides.com/demos/sample-videos/small.mp4" type="video/mp4"></video>'.
```

I checked in print_utils and I have changed all the expected behavior of the video test cases. @RFSH / @jrchudy could you please take a look at this for me? Once this test case passes, we are good to go. Everything else has passed.